ecommerce/enterprise_coupons/views.py

```
@@ -0,0 +1,39 @@
+"""
+Views for enterprise coupons application.
+"""
+
+import logging
+
+from django.shortcuts import redirect
+from ecommerce.enterprise_coupons.exceptions import CatalogueNotFoundError, CouponError, CouponNotFoundError
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import status
+
+from enterprise_coupons.utils import  get_enterprise_catalog_id_from_coupon, get_build_mfe_base_url
+
+logger = logging.getLogger(__name__)
+
+
+class CouponRedirectView(APIView):
+    def get(self, request, coupon_code, *args, **kwargs):
+        try:
+            catalog_id = get_enterprise_catalog_id_from_coupon(coupon_code)
+            mfe_url = get_build_mfe_base_url(catalog_id)
+            return redirect(mfe_url)
+        except CouponNotFoundError:
+            return Response(
+                {'error': 'Coupon code not found'},
+                status=status.HTTP_404_NOT_FOUND
+            )
+        except CatalogueNotFoundError:
+            return Response(
+                {'error': 'No catalogue available for this coupon'},
+                status=status.HTTP_404_NOT_FOUND
+            )
+        except CouponError as e:
+            logger.error(f"Coupon error: {str(e)}")
+            return Response(
+                {'error': 'Invalid coupon'},
+                status=status.HTTP_400_BAD_REQUEST
+            )
```

There is no default return for this view, we should have one as all views should return an HTTP response. @daviidco

---

The view always returns either `redirect(mfe_url) `on success or a Response from the exception handlers. A catch-all except Exception would hide unexpected errors that Django's middleware should handle with proper logging and tracebacks. @Squirrel18

---

So I guess in case of an unexpected error, you are expecting this view to fail with a 500 error, correct? if so, I think that's ok, but I would leave this decision somewhere in a doctring. @daviidco

---

Done

---

I was checking and the error returned by Django is this: `The view ... didn't return an HttpResponse object. It returned None instead`. So it's not giving us anything valuable related to the actual error, it's just saying that we didn't place a HttpResponse, so I don't think this will be an acceptable behavior, so in this case if you think we still need to return a 500, I will prefer to do it explicit and not implicit. I would even recommend to return a 404 and log the actual error internally so we can debugged it, and avoid giving much information about this endpoint to a bad actor. @daviidco

---

Done @Squirrel18